src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -639,11 +639,69 @@ def _process_batch_upload(
     return upload_results
 
 
+def _download_assembly_fasta_file(
+        asu_client: AssemblyUtil,
+        genome_obj_info: list[Any],
+        coll_src_assembly_path: str,
+        job_data_dir: Path | str,
+) -> Path:
+    """
+    Download assembly fasta file from the workspace using AssemblyUtil.
+    Copy the assembly file from the job directory to the collection source directory.
+    """
+    try:
+        assembly_upa = genome_obj_info[10]["Assembly Object"]
+    except KeyError:
+        raise ValueError(f"Genome object {genome_obj_info[1]} does not have an assembly object linked to it")
+    containerized_assembly_file = Path(asu_client.get_assembly_as_fasta({"ref": assembly_upa})["path"])
+    # remove prefix of the assembly_path
+    relative_assembly_path = containerized_assembly_file.relative_to(_JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER)
+    local_assembly_file = Path(job_data_dir) / relative_assembly_path
+
+    loader_helper.create_hardlink_between_files(coll_src_assembly_path,
+                                                local_assembly_file)
+
+    if not os.path.exists(coll_src_assembly_path):
+        raise ValueError(f"Retrieving assembly file from Genome {genome_obj_info[1]} failed.")
+
```

Comment:
Can this actually happen? I think the hardlink would've failed before this error

Reply:
hmmm. the last call in hardlink is `os.link(target_file, new_file)`. In theory, this shouldn't happen. But just in case? or do you think removing this check makes more sense?

Reply:
What happens if target_file doesn't exist in the link?

Reply:
if target_file doesn't exist, `os.path.samefile(target_file, new_file)` will throw an error in `create_hardlink_between_files`.

Reply:
So it seems like that raise line is impossible to reach

Reply:
👍 removed the condition.

Reply:
:+1: